test/addons/symlinked-module/submodule.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+require('../../common');
+const path = require('path');
+const assert = require('assert');
+
+// jshint unused:false
+module.exports.test = function test(bindingDir) {
+// jshint unused:true
+  var mod = require(path.join(bindingDir, 'binding.node'));
+  assert(mod != null);
+  assert(mod.hello() == 'world');
```

Comment:
Yeah, there are several issues I was noticing with this commit. I modified it just enough to get it running but hadn't gone through a detailed review yet.

@saper .. If you don't mind I'm going to go through and clean these up. I'll keep your name on the commit tho.